add support for binding responsive styles in gen1 react sdk

steve8708 opened the pull request:
currently gen2 SDKs support binding to responsive styles, like 

```js
{ bindings: { 'responsiveStyles.medium.color': 'state.foo'}
```

but not gen1, this fixes that

samijaber reviewed (commented):
oh, this logic fixes responsive styles? how are they broken exactly? in SSR? 

You initially described this same code as solving SSR of styles, but when Clyde added tests it still didn't address that so he made a different change: https://github.com/BuilderIO/builder/pull/3563

steve8708 commented:
Where was SSR mentioned? This ( https://github.com/BuilderIO/builder/pull/3485 ) was the original PR, I didn't realize we removed this code that I added (was supposed to get in pre launch). The Pr title and description were

> 
> ​ add support for responsive styles in gen1 react sdk and memoize
> processBlock
> 
> 

> 
> ​
> 
> 

> 
> ​we don't use the post-processed responsiveStyles object that bindings
> influence, but instead the preprocessed one
> 
> 

> 
> ​
> 
> 

> 
> ​gen2 supports this, but not gen1
> 
> 

tl;dr you could not bind to responsiveStyles before, but now you can

On Wed, Nov 13, 2024 at 11:08 AM, Sami Jaber < ***@***.*** > wrote:

> 
> 
> 
> ***@***.**** commented on this pull request.
> 
> 
> 
> oh, this logic fixes responsive styles? how are they broken exactly? in
> SSR?
> 
> 
> 
> You initially described this same code as solving SSR of styles, but when
> Clyde added tests it still didn't address that so he made a different
> change: #3563 ( https://github.com/BuilderIO/builder/pull/3563 )
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/BuilderIO/builder/pull/3732#pullrequestreview-2434191088
> ) , or unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AAGOEAYJJEPNZACL45GSE5L2AOPUXAVCNFSM6AAAAABRXE5NV6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMZUGE4TCMBYHA
> ).
> You are receiving this because you authored the thread. Message ID: <BuilderIO/builder/pull/3732/review/2434191088
> @ github. com>
> 
> 
>

samijaber commented:
> maybe just miscommunication somewhere?

Seems to be the case. Sorry this didn't make it in sooner.

steve8708 commented:
all good - best time to plant a tree is 10 years ago, second best time is today! thanks for the quick review
